src/Config/InfectionConfig.php

```diff
@@ -40,6 +40,8 @@
 
 /**
  * @internal
+ *
+ * @deprecated Should use {@see \Infection\Configuration\Configuration} instead
```

Comment:
Should be able to remove this class in this PR

Reply:
Thinking about it again: the PR is big enough as it is. It can be done later in a separate PR